libcst/codemod/visitors/_add_imports.py

```diff
@@ -169,14 +205,14 @@ def __init__(
             for module in sorted(from_imports_aliases)
         }
 
-        # Track the list of imports found in the file
-        self.all_imports: List[Union[libcst.Import, libcst.ImportFrom]] = []
+        # Track the list of imports found at the top of the file
+        self.all_imports: Set[Union[libcst.Import, libcst.ImportFrom]] = set()
```

Comment:
Unfortunately the List->Set is a breaking change in the API, let's avoid it unless it's absolutely necessary.